src/Component.js

```diff
@@ -0,0 +1,67 @@
+/* Represents a component of a vector */
+class Component
+{
+	constructor(length, maxRep, jump, hop, promote)
+	{
+		this.length = length;
+		this.maxRep = maxRep;
+		this.jump = jump;
+		this.hop = hop;
+		this.promote = promote;
+	}
+	
+	static Create(compStr, globalStr)
```

Comment:
I tend to lean on the side of verbosity when it comes to variable names. In this case, I would go for `componentString` and `globalString`. Small concern but we should aim for consistency on whatever we go with.